### ISSUE TITLE
compiler error with json serialization (Error: internal error: invalid kind for lastOrd(tyLent))

### ISSUE THREAD

**Issue by brentp** [nim-lang/Nim#15025]
This code:
```Nim
type Data* = object
  position*: seq[array[2, int]]

when isMainModule:

  import json
  var x: Data
  echo %x
```

generates this error:
```
Hint: used config file '/home/brentp/src/nim/config/nim.cfg' [Conf]
Hint: used config file '/home/brentp/src/nim/config/config.nims' [Conf]
...................
Error: internal error: invalid kind for lastOrd(tyLent)
No stack traceback available
To create a stacktrace, rerun compilation with './koch temp c <file>', see https://nim-lang.github.io/Nim/intern.html#debugging-the-compiler for details
```

**Comment by brentp** [comment]
updated to simplify. it seems `seq[array...]` is the problem.

**Comment by Araq** [comment]
Fixed in commit 80d3ef995282ec2e37968feedf7914ade6f2e62f
